Havtorn/Source/Engine/ECS/Systems/RenderSystem.cpp

```diff
@@ -56,42 +58,47 @@ namespace Havtorn
 			const I64 materialCompIndex = staticMeshComponent->Entity->GetComponentIndex(EComponentType::MaterialComponent);
 			auto& materialComp = materialComponents[materialCompIndex];
 
-			if (!directionalLightComponents.empty())
-			{
-				std::array<Ref<SComponent>, static_cast<size_t>(EComponentType::Count)> components;
-				components[static_cast<U8>(EComponentType::TransformComponent)] = transformComp;
-				components[static_cast<U8>(EComponentType::StaticMeshComponent)] = staticMeshComponent;
-				components[static_cast<U8>(EComponentType::DirectionalLightComponent)] = directionalLightComponents[0];
-				SRenderCommand command(components, ERenderCommandType::ShadowAtlasPrePassDirectional);
-				RenderManager->PushRenderCommand(command);
-			}
+			if (!RenderManager->IsStaticMeshInInstancedRenderList(staticMeshComponent->Name)) // if static, if instanced
+			{		
+				if (!directionalLightComponents.empty())
+				{
+					std::array<Ref<SComponent>, static_cast<size_t>(EComponentType::Count)> components;
```

Comment:
Is it not possible to turn this segment into a function, there are onlya few differences between directional, point and spot.

Reply:
Referring to ex: lines 65-70 == 75-80 == 85-90

Reply:
It is, wanted to do a first pass before checking whether function calls will be too detrimental or not

Reply:
There is a bug hiding as well so was gonna look at it next